include/secp256k1_silentpayments.h

```diff
@@ -40,6 +40,59 @@ typedef struct {
     size_t index;
 } secp256k1_silentpayments_recipient;
 
+/** Create Silent Payment outputs for recipient(s).
+ *
+ * Given a list of n private keys a_1...a_n (one for each silent payment
+ * eligible input to spend), a serialized outpoint, and a list of recipients,
+ * create the taproot outputs:
+ *
+ * a_sum = a_1 + a_2 + ... + a_n
+ * input_hash = hash(outpoint_smallest || (a_sum * G))
+ * taproot_output = B_spend + hash(a_sum * input_hash * B_scan || k) * G
+ *
+ * If necessary, the private keys are negated to enforce the right y-parity.
+ * For that reason, the private keys have to be passed in via two different parameter
+ * pairs, depending on whether they seckeys correspond to x-only outputs or not.
+ *
+ *  Returns: 1 if shared secret creation was successful. 0 if an error occured.
+ *   Args:                 ctx: pointer to a context object
+ *    Out:   generated_outputs: pointer to an array of pointers to xonly pubkeys, one per recipient.
+ *                              The order of outputs here matches the original ordering of the
+ *                              recipients array.
+ *     In:          recipients: pointer to an array of pointers to silent payment recipients,
```

Comment:
Good question. I would argue no, since the caller is passing the array in and then never using it again. The fact that the array is sorted in place is irrelevant to the caller since they shouldn't be reading those values again, anyways. But maybe there is a more strict definition of In/Out, that if the function can modify the parameter it is an Out param?

Reply:
I don't think there's a definition for this case and I don't remember a precedent. "In" seems fine to me if the modified argument is not supposed to be read again. The doc also mentions already that the array is modified ("The recipient array will be sorted in place").

Reply:
Going to leave it as "In" for now. I think it makes it more clear to the caller that they are supposed to read the `recipients` array after the function call.